onnxruntime/test/ir/flatbuffer_utils_test.cc

```diff
@@ -0,0 +1,409 @@
+
+  // 16-bit. 81 elements so we're 2 bytes past 4 byte alignment
+  initializers.emplace_back(
+      CreateInitializer<int16_t>("tensor_16", ONNX_NAMESPACE::TensorProto_DataType_INT16, {9, 9}));
+
+  // string (should not use external)
+  initializers.emplace_back(
+      CreateInitializer<std::string>("tensor_string", ONNX_NAMESPACE::TensorProto_DataType_STRING, {2, 2}));
+
+  // 32-bit, 64 elements
+  initializers.emplace_back(
+      CreateInitializer<float>("tensor_f32", ONNX_NAMESPACE::TensorProto_DataType_FLOAT, {8, 8}));
+
+  // 8-bit. 129 elements so we're 1 byte past 4 or 8 byte alignment
+  initializers.emplace_back(
+      CreateInitializer<uint8_t>("tensor_8", ONNX_NAMESPACE::TensorProto_DataType_UINT8, {3, 43}));
+
+  // 64-bit, 36 elements
+  initializers.emplace_back(
+      CreateInitializer<int64_t>("tensor_64", ONNX_NAMESPACE::TensorProto_DataType_INT64, {6, 6}));
+
+  // small (should not use external)
+  initializers.emplace_back(
+      CreateInitializer<int32_t>("tensor_32_small", ONNX_NAMESPACE::TensorProto_DataType_INT32, {2, 2}));
+
+  return initializers;
+}
+
+#ifdef ENABLE_TRAINING_APIS
+std::vector<ONNX_NAMESPACE::TensorProto> CreateInitializersNoString() {
+  std::vector<ONNX_NAMESPACE::TensorProto> initializers;
+  // create data of various sizes. order is chosen to require padding between most but not all
+  // assuming our writer aligns to 4 bytes unless it's 64-bit data (which is aligned to 8 bytes)
+  // buffer: <16-bit><pad 2><32-bit><8-bit><pad 7><64-bit>
+  // need 128 bytes to write to external data
+
+  // 16-bit. 81 elements so we're 2 bytes past 4 byte alignment
+  initializers.emplace_back(
+      CreateInitializer<int16_t>("tensor_16", ONNX_NAMESPACE::TensorProto_DataType_INT16, {9, 9}));
+
+  // 32-bit, 64 elements
+  initializers.emplace_back(
+      CreateInitializer<float>("tensor_f32", ONNX_NAMESPACE::TensorProto_DataType_FLOAT, {8, 8}));
+
+  // 8-bit. 129 elements so we're 1 byte past 4 or 8 byte alignment
+  initializers.emplace_back(
+      CreateInitializer<uint8_t>("tensor_8", ONNX_NAMESPACE::TensorProto_DataType_UINT8, {3, 43}));
+
+  // 64-bit, 36 elements
+  initializers.emplace_back(
+      CreateInitializer<int64_t>("tensor_64", ONNX_NAMESPACE::TensorProto_DataType_INT64, {6, 6}));
+
+  // small (should not use external)
+  initializers.emplace_back(
+      CreateInitializer<int32_t>("tensor_32_small", ONNX_NAMESPACE::TensorProto_DataType_INT32, {2, 2}));
+
+  return initializers;
+}
+#endif // ENABLE_TRAINING_APIS
+
+template <typename T>
+std::vector<T> ConvertRawDataToTypedVector(ONNX_NAMESPACE::TensorProto initializer) {
+  std::vector<T> data;
+  data.resize(initializer.raw_data().size() / sizeof(T));
+  memcpy(data.data(), initializer.raw_data().data(), initializer.raw_data().size());
+  return data;
+}
+
+#define ASSERT_EQ_FB_TENSORPROTO_VECTORFIELD(EXPECTED, ACTUAL, FIELD) \
+  ASSERT_EQ(EXPECTED.FIELD.size(), ACTUAL.FIELD.size());              \
```

Comment:
FYI, for macros, usually good to put the arguments which might be expressions within parentheses to avoid unexpected parsing.
```suggestion
  ASSERT_EQ((EXPECTED).FIELD.size(), (ACTUAL).FIELD.size());              \
```